parts/Join-SwarmMode-cluster.ps1

```diff
@@ -0,0 +1,335 @@
+    Remove-Item $zipfile
+}
+
+function Update-DockerServiceRecoveryPolicy()
+{
+    $dockerReady = $false
+    $startTime = Get-Date
+    
+    # wait until the service exists
+    while (-not $dockerReady)
+    {
+        if (Get-Service $global:DockerServiceName -ErrorAction SilentlyContinue)
+        {
+            $dockerReady = $true
+        }
+        else 
+        {
+            $timeElapsed = $(Get-Date) - $startTime
+            if ($($timeElapsed).TotalMinutes -ge 5)
+            {
+                Write-Log "Unable to find service $global:DockerServiceName within 5 minutes."
+                break
+            }
+            Write-Log "failed to find $global:DockerServiceName, sleeping for 5 seconds"
+            Start-Sleep -sec 5
+        }
+    }
+    
+    Write-Log "Updating docker restart policy, to ensure it restarts on error"
+    $services = Get-WMIObject win32_service | Where-Object {$_.name -imatch $global:DockerServiceName}
+    foreach ($service in $services)
+    {
+        sc.exe failure $service.name reset= 86400 actions= restart/5000
+    }
+}
+
+# Open Windows Firewall Ports Needed
+function Open-FirewallPorts()
+{
+    $tcpports = @(80,443,2375,8080,2377,7946,4789)
+    foreach ($tcpport in $tcpports)
+    {
+        $netsh = "netsh advfirewall firewall add rule name='Open Port $tcpport' dir=in action=allow protocol=TCP localport=$tcpport"
+        Write-Log "enabling port with command $netsh"
+        Invoke-Expression -Command:$netsh
+    }
+
+    $udpports = @(7946,4789)
+    foreach ($udpport in $udpports)
+    {
+        $netsh = "netsh advfirewall firewall add rule name='Open Port $udpport' dir=in action=allow protocol=UDP localport=$udpport"
+        Write-Log "enabling port with command $netsh"
+        Invoke-Expression -Command:$netsh
+    }
+}
+
+# Update Docker Config to have cluster-store=consul:// address configured for Swarm cluster.
+function Write-DockerDaemonJson()
+{
+    $dataDir = $env:ProgramData
+
+    # create the target directory
+    $targetDir = $dataDir + '\docker\config'
+    if(!(Test-Path -Path $targetDir )){
+        New-Item -ItemType directory -Path $targetDir
+    }
+
+    Write-Log "Delete key file, so that this node is unique to swarm"
+    $keyFileName = "$targetDir\key.json"
+    Write-Log "Removing $($keyFileName)"
+    if (Test-Path $keyFileName) {
+      Remove-Item $keyFileName
+    }
+
+    Write-Log "Configure Docker Engine to accept incoming connections on port 2375"
+    $OutFile = @"
+{
+    "hosts": ["tcp://0.0.0.0:2375", "npipe://"]
+}
+"@
+
+    $OutFile | Out-File -encoding ASCII -filepath "$targetDir\daemon.json"
+}
+
+function Join-Swarm()
+{
+    $currentRetry = 0;
+    $success = $false;
+    $getTokenCommand = "docker -H $($SwarmMasterIP):2375 swarm join-token -q worker"
+    $swarmmodetoken;
+
+    do {
+        try
+        {
+            Write-Log "Executing [$getTokenCommand] command...."
+            <#& $swarmmodetoken#>

```

Comment:
This command gets the token for the worker from the master. Then in the last command at the end of this function uses that token join the Swarm.